jupyterlab_server/handlers.py

```diff
@@ -292,6 +304,34 @@ def add_handlers(web_app, config):
         handlers.append((
             workspace_api_path, WorkspacesHandler, workspaces_config))
 
+    # Handle local listings.
+    if config.listings_dir:
+
+        settings_config = web_app.settings.get('config', {}).get('LabServerApp', {})
+        blacklist_uris = settings_config.get('blacklist_uris', '')
```

Comment:
Since you specify defaults for these settings values above, can you just use the getitem and assume they will exist? Seems unnecessary to put defaults in two places.

Reply:
jupyter traits, and specifically how they are implemented in jupyterlab / jupyterlab-server are  strange. The defaults are stated for the developer, but they do not work as expected in my understanding. I have spent a lot of time to debug and making sure the pieces were correctly setup, also discussed with Zach. The new server extensions will make it more clear, but for now, I propose to keep as it is implemented.